spec/lib/miq_pglogical/miq_pglogical_subscriber_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'miq_pglogical'
+
+describe MiqPglogicalSubscriber do
+  let(:subscriptions) do
+    [
+      {
+        :dbname => "test_db",
+        :host   => "example.com"
+      },
+      {
+        :name   => "subscription_db_example_com",
+        :dbname => "test_db2",
+        :host   => "db.example.com"
+      }
+    ]
+  end
+
+  let(:subscription_names) { %w(subscription_db_example_com subscription_example_com) }
+
+  before do
+    MiqServer.seed
+  end
+
+  describe "#configured_subscriptions" do
+    it "returns an empty array if no subscriptions are configured" do
+      expect(subject.configured_subscriptions).to eq([])
+    end
+
+    context "with subscriptions" do
+      before do
+        c = MiqServer.my_server.get_config
+        c.config.store_path(*described_class::SETTINGS_PATH, :subscriptions, subscriptions)
+        c.save
+      end
+
+      it "names subscriptions without a name" do
+        subject.configured_subscriptions.each do |s|
+          expect(s[:name]).to eq("subscription_#{s[:host].gsub(/\.|-/, "_")}")
```

Comment:
I think it would be better to hardcode the expected value here, instead of copying the gsub code, so you can be sure your regex is tested properly.